frontend/src/components/Dropdown.tsx

```diff
@@ -0,0 +1,47 @@
+import { createSignal } from "solid-js";
+
+interface optionInfo {
+    name: string;
+    value: string;
+    id: string;
+}
+
+interface menuProps {
+    menuOption: optionInfo[];
+    placeholder?: string;
+}
+
+export function Dropdown({menuOption}: menuProps) {
+    const [menuIsOpen, setMenuIsOpen] = createSignal(true);
+    const [selected, setSelected] = createSignal("")
+
+    function toggleMenu() {
```

Comment:
Since this function only calls a different function, you can instead call the inner function immediately in the component:
```jsx
 <div 
class="relative trigger menu bg-[url('assets/images/caretDown.svg')] bg-no-repeat bg-[left_11rem_top_57%] bg-[length:1rem_1rem] hover:cursor-pointer" 
onClick={(prev) => setMenuIsOpen(!prev)}
>
```